sources/packages/backend/apps/queue-consumers/src/services/cas-supplier/cas.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from "@nestjs/common";
+import {
+  CASAuthDetails,
+  CASSupplierResponse,
+} from "../../processors/schedulers/cas-integration/models/cas-supplier-response.dto";
+import { AxiosRequestConfig } from "axios";
+import { HttpService } from "@nestjs/axios";
+import { InjectRepository } from "@nestjs/typeorm";
+import { CASSupplier } from "@sims/sims-db";
+import { CASIntegrationConfig, ConfigService } from "@sims/utilities/config";
+import { Repository } from "typeorm";
+
+@Injectable()
+export class CASService {
+  private readonly casIntegrationConfig: CASIntegrationConfig;
+  constructor(
+    config: ConfigService,
+    private readonly httpService: HttpService,
+    @InjectRepository(CASSupplier)
+    private readonly casSupplierRepo: Repository<CASSupplier>,
+  ) {
+    this.casIntegrationConfig = config.casIntegration;
+  }
+  /**
+   * Request to login on CAS API and return CAS auth details with the token used for authentication in all other requests.
+   * @returns CAS auth details.
+   */
+  public async casLogon(): Promise<CASAuthDetails> {
+    const url = `${this.casIntegrationConfig.baseUrl}/oauth/token`;
+    const auth = Buffer.from(
+      `${this.casIntegrationConfig.clientId}:${this.casIntegrationConfig.clientSecret}`,
+    ).toString("base64");
+    const headers = {
+      "Content-Type": "application/x-www-form-urlencoded",
+      Authorization: `Basic ${auth}`,
+    };
+    const body = new URLSearchParams();
+    body.append("grant_type", "client_credentials");
+    const config: AxiosRequestConfig = {
+      headers,
+    };
+    const response = await this.httpService.axiosRef.post(
+      url,
+      body.toString(),
```

Comment:
For Keycloak integration `stringify` was used. Can we keep the same?
https://github.com/bcgov/SIMS/blob/4dde03d07b37a188ce39381f986f5a1452ae90e8/sources/packages/backend/libs/auth/src/services/keycloak/keycloak.service.ts#L118